### PR TITLE
snap: add tests for multiple commands

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Adds tests for handling multiple snaps in the same command.

##### ISSUE TYPE
- Test Pull Request

##### COMPONENT NAME
snap

### CODE REVIEW

**Comment by felixfontein** [comment]
@russoz what's the state on this?

**Comment by russoz** [comment]
This is out there in the cold, crying alone in the dark.

I am working on a couple of other things first, then will come back to this.

**Comment by russoz** [comment]
Added `become: true` to the tests to make it easier to test locally with vagrant.

File: tests/integration/targets/snap/tasks/main.yml [hunk 1]
```diff
@@ -191,3 +192,42 @@
...
+        state: present
+      register: install_two_again
+
+    - name: Remove packages (hello-world & uhttpd)
+      community.general.snap:
+        name:
+          - hello-world
+          - uhttpd
+        state: absent
+      register: install_two_remove
```
Comment (felixfontein) [inline tests/integration/targets/snap/tasks/main.yml:218]: How about also removing both again?

Comment (russoz) [reply tests/integration/targets/snap/tasks/main.yml:218]: sure, why not :-)

**Comment by russoz** [comment]
@felixfontein I am trying to get the vagrant-based test-runs going here, and for this one (snap) I had it working by running the test straight as `vagrant` user, but using the `become: true`. I tried to do something similar to #5144 but it complained that it was missing the `--allow-root` parameter, because of the `needs/root` in `aliases`.

Couple of thoughts here:
1. If we are to use `needs/root`, then `snap` should have that flag as well, because root is required to un/install snaps. Right? Or was there any particular reason not to set that flag in this test?
2. If we used `become: true` instead of `needs/root`, wouldn't that achieve the same result using a standard construct from Ansible instead of a construct invented exclusively for the CI? I mean, the tests worked in this PR here. I feel like I am missing some part of the picture, the flag must be used for something else as well.

**Comment by felixfontein** [comment]
> If we are to use needs/root, then snap should have that flag as well, because root is required to un/install snaps. Right? Or was there any particular reason not to set that flag in this test?

It's probably not set because it also works without it in CI... But yes, it should be set there as well.

> If we used `become: true` instead of `needs/root`, wouldn't that achieve the same result using a standard construct from Ansible instead of a construct invented exclusively for the CI? I mean, the tests worked in this PR here. I feel like I am missing some part of the picture, the flag must be used for something else as well.

I'm not sure whether that works well with ansible-test. The `needs/root` is mainly to make sure that you don't accidentally run tests that need root on your system without virtualization.

File: tests/integration/targets/setup_snap/handlers/main.yml [hunk 2]
```diff
@@ -7,3 +7,4 @@
   package:
     name: "{{ snap_packages }}"
     state: absent
+  become: true
```
Comment (felixfontein) [inline tests/integration/targets/setup_snap/handlers/main.yml:10]: I would avoid using `become: true`.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for improving the tests!

**Pull request merged**